Date: Thu, 19 Mar 2009 09:58:32 -0700
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: KABI: fix for struct seq_file change

2.6.27.20 changes the seq_file structure, adding the read_pos field.

No modules "should" be creating the seq_file structure, so moving it to
the end, and taking it out of the kernel symbol check, should resolve
the issue.

Hopefully.

Crossing my fingers.

Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 include/linux/seq_file.h |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/include/linux/seq_file.h
+++ b/include/linux/seq_file.h
@@ -19,11 +19,13 @@ struct seq_file {
 	size_t from;
 	size_t count;
 	loff_t index;
-	loff_t read_pos;
 	u64 version;
 	struct mutex lock;
 	const struct seq_operations *op;
 	void *private;
+#ifndef __GENKSYMS__
+	loff_t read_pos;
+#endif
 };
 
 struct seq_operations {
